extensions/ros2/launch

```diff
@@ -13,6 +13,13 @@ function source_with_prefix() {
 # Save off parameters, the sourced setup scripts may manipulate them.
 original_args=("$@")
 
+if [ -d "$SNAP/opt/ros/underlay_ws/opt/ros/$ROS_DISTRO" ]; then
+  source_with_prefix "$SNAP/opt/ros/underlay_ws/opt/ros/$ROS_DISTRO"
```

Comment:
Same about `source_with_prefix`. It would be simpler if it would check itself for the existence and simply log in case it doesn't exist no?